_source/_posts/2022-05-02-angular-microfrontend-auth.md

```diff
@@ -0,0 +1,443 @@
+          shareReplay()
+      );
+  
+  public name$: Observable<string> = this.oktaStateService.authState$
+      .pipe(
+          filter(authState => !!authState && !!authState.isAuthenticated),
+          map(authState => authState.idToken?.claims.name ?? '')
+      );
+
+  constructor(private oktaStateService: OktaAuthStateService, @Inject(OKTA_AUTH) private oktaAuth: OktaAuth) { }
+
+  public async signIn(): Promise<void> {
+    await this.oktaAuth.signInWithRedirect();
+  }
+
+  public async signOut(): Promise<void> {
+    await this.oktaAuth.signOut();
+  }
+}
+```
+
+We need to add the click handlers for the sign-in and sign-out buttons. Open `app.component.html` in the `shell` project. Update the code for **Sign In** and **Sign Out** buttons as shown.
+
+```html
+<li>
+  <button *ngIf="(isAuthenticated$ | async) === false; else logout"
+    class="flex items-center transition ease-in delay-150 duration-300 h-10 px-4 rounded-lg hover:border hover:border-sky-400"
+    (click)="signIn()"
+  >
+    <span class="material-icons-outlined text-gray-500">login</span>
+    <span>&nbsp;Sign In</span>
+  </button>
+
+    <ng-template #logout>
+      <button 
+        class="flex items-center transition ease-in delay-150 duration-300 h-10 px-4 rounded-lg hover:border hover:border-sky-400"
+        (click)="signOut()"
+      >
+        <span class="material-icons-outlined text-gray-500">logout</span>
+        <span>&nbsp;Sign Out</span>
+      </button>
+  </ng-template>
+</li>
+```
+
+Try running the project using `npm run run:all`. Now you'll be able to sign in and sign out. And when you sign in, a new button for **Profile** shows up. Nothing happens when you click it, but we're going to create a new remote, connect it to the host, and share the authenticated state here next!
+
+## Create a new Angular application
+
+Now you'll have a chance to dive in a little deeper to see how a micro-frontend remote connects to the host by creating a micro-frontend app to show the authenticated user's profile information. Stop serving the project and create a new Angular application in the project by running the following command in the terminal:
+
+```shell
+ng generate application mfe-profile --routing --style css --inline-style --skip-tests
+```
+
+With this Angular CLI command you
+1. Generated a new application named `mfe-profile` which includes a module and a component
+2. Added a separate routing module to the application
+3. Defined the CSS styles to be inline in the components
+4. Skipped creating associated test files for the initial component
+
+You'll now create a component for the default route, `HomeComponent`, and a module to house the micro-frontend. We could wire up the micro-frontend to only use a component instead of a module, and for a profile view, a component will cover our needs, but we'll use a module so you can see how each micro-frontend can grow as the project evolves. Run the following two commands in the terminal:
```

Comment:
```suggestion
You'll now create a component for the default route, `HomeComponent`, and a module to house the micro frontend. We could wire up the micro frontend to only use a component instead of a module. In fact, a component will cover our needs for a profile view, , but we'll use a module so you can see how each micro frontend can grow as the project evolves. Run the following two commands in the terminal:
```
removed hyphens. Made a long complex sentence into 2 sentences.